include/vkwrap/vkwrap/mman.h

```diff
@@ -0,0 +1,538 @@
+
+    VmaDetailedStatistics getTotalStats() const
+    {
+        VmaTotalStatistics stats{};
+        vmaCalculateStatistics( m_vma, &stats );
+
+        return stats.total;
+    } // getTotalStats
+
+    OneTimeCommand& getCommand() { return m_cmd; }
+
+    static constexpr VmaAllocationCreateInfo k_buffer_alloc_create_info{
+        .flags = VMA_ALLOCATION_CREATE_HOST_ACCESS_RANDOM_BIT,
+        .usage = VMA_MEMORY_USAGE_AUTO };
+
+    static constexpr VmaAllocationCreateInfo k_image_alloc_create_info{
+        .usage = VMA_MEMORY_USAGE_AUTO,
+        .requiredFlags = VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT };
+
+    static constexpr VmaVulkanFunctions k_vulkan_functions = {
+        .vkGetInstanceProcAddr = &vkGetInstanceProcAddr,
+        .vkGetDeviceProcAddr = &vkGetDeviceProcAddr };
+
+    static OneTimeCommand createCommand( vk::Device device, vk::CommandPool cmd_pool, vk::Queue queue )
+    {
+        vk::CommandBufferAllocateInfo alloc_info{
+            .commandPool = cmd_pool,
+            .level = vk::CommandBufferLevel::ePrimary,
+            .commandBufferCount = 1 };
+
+        // clang-format off
+        vk::UniqueCommandBuffer cmd_buffer{ 
+            std::move( device.allocateCommandBuffersUnique( alloc_info )[ 0 ] ) };
+        // clang-format on
+
+        return { std::move( cmd_buffer ), queue };
+    } // createCommandBuffer
+
+  public:
+    Mman(
+        vkwrap::VulkanVersion version,
+        vk::Instance instance,
+        vk::PhysicalDevice physical_device,
+        vk::Device logical_device,
+        vk::Queue queue,
+        vk::CommandPool cmd_pool )
+        : m_vma{ VK_NULL_HANDLE },
+          m_queue{ queue },
+          m_cmd{ createCommand( logical_device, cmd_pool, queue ) }
+    {
+        VmaAllocatorCreateInfo create_info{
+            .physicalDevice = physical_device,
+            .device = logical_device,
+            .pVulkanFunctions = &k_vulkan_functions,
+            .instance = instance,
+            .vulkanApiVersion = utils::toUnderlying( version ) };
+
+        VkResult result = vmaCreateAllocator( &create_info, &m_vma );
+        vk::resultCheck( vk::Result{ result }, "Mman: allocator creation error." );
+    } // Mman
+
+    ~Mman() { vmaDestroyAllocator( m_vma ); }
+
+    Mman( const Mman& ) = delete;
+    Mman( Mman&& ) = delete;
+    Mman& operator=( const Mman& ) = delete;
+    Mman& operator=( Mman&& ) = delete;
+
+    vk::Buffer create( const vk::BufferCreateInfo& create_info )
+    {
+        VkBuffer buffer{};
+        VmaAllocation allocation{};
+
+        VkResult result = vmaCreateBuffer(
+            m_vma,
+            &static_cast<const VkBufferCreateInfo&>( create_info ),
+            &k_buffer_alloc_create_info,
+            &buffer,
+            &allocation,
+            nullptr );
+
+        vk::resultCheck( vk::Result{ result }, "Mman: buffer allocation error." );
+
+        vk::Buffer buffer_hpp{ buffer };
```

Comment:
No need of `buffer_hpp` variable, the vk::Buffer has implicit constructor, so you can use `buffer`. Same with `image_hpp` and `image`

Reply:
Agree.